# **Live Code 1**
### **HACTIV8 Full Time Data Science Phase 0**

# **1. Perkenalan**

> Nama  : Ropiudin

> Batch : FTDS OO2-HCK

> Live Code : Live Code 1

# **2. Judul/Penanda Soal**

> Keterangan Soal

R=V/I

P=VI

R adalah Hambatan dalam satuan Ohm

P adalah Daya dalam Watt

I adalah kuat arus dalam satuan Ampere

V adalah tegangan dalam satuan Volt
 > Soal
 > 1. Buatlah dua fungsi eksplisit masing-masing memuat perhitungan hambatan (R) dan daya (P) yang nantinya akan digunakan untuk perhitungan
 > 2. Buat kolom baru dengan nama kolom R/P berisikan hasil perhitungan daya/hambatan dengan kriteria untuk Kapasitas >=2.5 hitung daya, Kapasitas <2.5 hitung nilai hambatan. **(Hint: Gunakan for loop dan if statement untuk mengerjakan perintah nomor b, kamu bisa menggunakan list untuk menyimpan perhitungan sementara, setelahnya dapat dimasukkan ke kolom R/P)**.
 > 3. Hitunglah rata-rata nilai kapasitas untuk arus = 1 Ampere dan cycle ganjil.

# **3. Pustaka yang digunakan**

In [1]:
import pandas as pd

> Code di atas merupakan code untuk import library pandas yang akan di gunakan pada analisis 

# **4. Data Loading**

In [5]:
df = pd.read_csv('Data LC1.csv')
df

time(s)   V_m  Current (A)  Cycle   Cap
0            0  4.10          1.0      1  2.84
1            1  4.10          1.0      1  2.84
2            2  4.09          1.0      1  2.84
3            3  4.09          1.0      1  2.84
4            4  4.09          1.0      1  2.84
...        ...   ...          ...    ...   ...
90977    90977  2.51          1.0      9  2.77
90978    90978  2.50          1.0      9  2.77
90979    90979  2.50          1.0      9  2.77
90980    90980  2.50          1.0      9  2.77
90981    90981  2.50          1.0      9  2.77

[90982 rows x 5 columns]

> Code di atas merupakan code untuk import data atau membaca data dari tabel yang akan digunakan

In [6]:
df.head()

time(s)   V_m  Current (A)  Cycle   Cap
0        0  4.10          1.0      1  2.84
1        1  4.10          1.0      1  2.84
2        2  4.09          1.0      1  2.84
3        3  4.09          1.0      1  2.84
4        4  4.09          1.0      1  2.84

> Code df.head() di atas merupakan code untuk menampilkan tabel teratas yang bisa di tentukan jumlah baris yang mau di tampilkan pada variabel df

In [7]:
df.tail()

time(s)   V_m  Current (A)  Cycle   Cap
90977    90977  2.51          1.0      9  2.77
90978    90978  2.50          1.0      9  2.77
90979    90979  2.50          1.0      9  2.77
90980    90980  2.50          1.0      9  2.77
90981    90981  2.50          1.0      9  2.77

> Code df.tail() di atas merupakan code untuk menampilkan tabel terbawah yang bisa di tentukan jumlah baris yang mau di tampilkan pada variabel df

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90982 entries, 0 to 90981
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   time(s)      90982 non-null  int64  
 1   V_m          90982 non-null  float64
 2   Current (A)  90982 non-null  float64
 3   Cycle        90982 non-null  int64  
 4   Cap          90982 non-null  float64
dtypes: float64(3), int64(2)
memory usage: 3.5 MB


> Code df.info() merupakan code untuk menampil kan informasi terkait data yang kita punya, yang berisikan jumalah baris dan kolom pada data serta nama - nama kolom. Dan terdapat pula informasi missing values di setiap kolom pada data

In [9]:
df.describe()

time(s)       V_m  Current (A)     Cycle       Cap
count  90982.00  90982.00     9.10e+04  90982.00  90982.00
mean   45490.50      3.61     9.99e-01      4.98      2.81
std    26264.39      0.32     2.00e-03      2.58      0.02
min        0.00      2.50     9.93e-01      1.00      2.77
25%    22745.25      3.42     9.97e-01      3.00      2.79
50%    45490.50      3.64     9.99e-01      5.00      2.80
75%    68235.75      3.87     1.00e+00      7.00      2.82
max    90981.00      4.11     1.00e+00      9.00      2.84

> Code df.describe() merupakan code untuk meanmpilkan statistika dasar pada data seperti count, mean dll.

# **5. Answer**


1. Buatlah dua fungsi eksplisit masing-masing memuat perhitungan hambatan (R) dan daya (P) yang nantinya akan digunakan untuk perhitungan

In [45]:
def hambatan_R(V,I):
  hambatan_R = (V/I)
  return hambatan_R

In [46]:
def daya_P(V,I):
  daya_P = (V*I)
  return daya_P

2. Buat kolom baru dengan nama kolom R/P berisikan hasil perhitungan daya/hambatan dengan kriteria untuk Kapasitas >=2.5 hitung daya, Kapasitas <2.5 hitung nilai hambatan. **(Hint: Gunakan for loop dan if statement untuk mengerjakan perintah nomor b, kamu bisa menggunakan list untuk menyimpan perhitungan sementara, setelahnya dapat dimasukkan ke kolom R/P)**.

In [67]:
df['R/P'] = 0
for i in df.index:
  if df['V_m'][i]>=2.5:
    df['R/P'][i]=daya_P(df['V_m'][i],df['Current (A)'][i])
  else:
    df['R/P'][i]=hambatan_R(df['V_m'][i],df['Current (A)'][i])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [68]:
df

time(s)   V_m  Current (A)  Cycle   Cap  R/P
0            0  4.10          1.0      1  2.84    4
1            1  4.10          1.0      1  2.84    4
2            2  4.09          1.0      1  2.84    4
3            3  4.09          1.0      1  2.84    4
4            4  4.09          1.0      1  2.84    4
...        ...   ...          ...    ...   ...  ...
90977    90977  2.51          1.0      9  2.77    2
90978    90978  2.50          1.0      9  2.77    2
90979    90979  2.50          1.0      9  2.77    2
90980    90980  2.50          1.0      9  2.77    2
90981    90981  2.50          1.0      9  2.77    2

[90982 rows x 6 columns]

3. Hitunglah rata-rata nilai kapasitas untuk arus = 1 Ampere dan cycle ganjil.

In [69]:
New_1 = df.copy()
New_1

time(s)   V_m  Current (A)  Cycle   Cap  R/P
0            0  4.10          1.0      1  2.84    4
1            1  4.10          1.0      1  2.84    4
2            2  4.09          1.0      1  2.84    4
3            3  4.09          1.0      1  2.84    4
4            4  4.09          1.0      1  2.84    4
...        ...   ...          ...    ...   ...  ...
90977    90977  2.51          1.0      9  2.77    2
90978    90978  2.50          1.0      9  2.77    2
90979    90979  2.50          1.0      9  2.77    2
90980    90980  2.50          1.0      9  2.77    2
90981    90981  2.50          1.0      9  2.77    2

[90982 rows x 6 columns]

In [70]:
New_2 = New_1[
    (New_1['Current (A)'] == 1) &
    (New_1['Cycle'] % 2) == 1
    ]
New_2

time(s)   V_m  Current (A)  Cycle   Cap  R/P
15          15  4.08          1.0      1  2.84    4
19          19  4.08          1.0      1  2.84    4
20          20  4.08          1.0      1  2.84    4
21          21  4.08          1.0      1  2.84    4
31          31  4.08          1.0      1  2.84    4
...        ...   ...          ...    ...   ...  ...
90909    90909  2.64          1.0      9  2.77    2
90923    90923  2.62          1.0      9  2.77    2
90937    90937  2.59          1.0      9  2.77    2
90949    90949  2.56          1.0      9  2.77    2
90967    90967  2.53          1.0      9  2.77    2

[6830 rows x 6 columns]

In [71]:
Rata_rata = New_2[New_2['Current (A)'] == 1]['R/P'].mean()
Rata_rata

3.044216691068814

# **6. Hasil**

> Dari hasil proses analisis di atas di peroleh hasil nilai rata - rata kapasitas dari data dengan  nilai arus = 1 Ampere dan cycle ganjil yaitu sebesar 3.044216691068814